### PR TITLE
nvidia additions

### PR DESCRIPTION
**Pull request by dougg0k**
I just re-added in a small PR with additionals.

https://discuss.cachyos.org/t/random-white-box-flickering/3254/2

https://discuss.cachyos.org/t/resizable-bar-stay-disabled/3154

### CODE REVIEW

**Comment by Edgars-Cirulis** [comment]
Looks similar  https://aur.archlinux.org/cgit/aur.git/tree/0004-NVreg-Improvements.patch?h=nvidia-dkms-performance

**Comment by 1Naim** [comment]
> Looks similar https://aur.archlinux.org/cgit/aur.git/tree/0004-NVreg-Improvements.patch?h=nvidia-dkms-performance

Yep, that's @ventureoo 😄

**Comment by ventureoo** [comment]
> Why not adding NVreg_EnablePCIeGen3=1 ? even though if it's not compatible it should do fallback

This is not required on desktop cards.  The default PCIe Gen 2 mode of operation is only for Quadro, Tesla, and NVS Kepler, which is not a common case. See: https://github.com/CachyOS/CachyOS-Settings/pull/3

File: usr/lib/modprobe.d/nvidia.conf [hunk 1]
```diff
@@ -1,35 +1,49 @@
...
+
+# Vreg_EnableStreamMemOPs=1 (Default 0) - Activates the support for
+# CUDA Stream Memory Operations in user-mode applications.
+options nvidia NVreg_EnableStreamMemOPs=1
+
+# Enable PowerMizer
+# Adaptive Frequencies on DC/AC
+# Adaptive Perf on DC
+# Maximum Perf on AC
+options nvidia NVreg_RegistryDwords="PowerMizerEnable=0x1;PerfLevelSrc=0x3333;PowerMizerDefault=0x2;PowerMizerDefaultAC=0x1"
```
[the remark this answers is not in the document]
Comment (dougg0k) [reply usr/lib/modprobe.d/nvidia.conf:40]: It fixed an issue that I had, if you look at the first url. You sure? I never saw the issue again after applying it. And before it was always randomly there.

**Comment by dougg0k** [comment]
It's much more readable to have in each separated. And they work the same way. Your common practice here seems just because someone else did like that first.

File: usr/lib/modprobe.d/nvidia.conf [hunk 2]
```diff
@@ -1,35 +1,49 @@
...
 # NVreg_EnableGpuFirmware=0 (Default 1) - Disables GSP Firmware on NVIDIA's
 # closed source kernel modules. This option is ignored when using NVIDIA's
 # open kernel modules.
-#
+
+# Disable GSP Firmware on NVIDIA's closed source kernel modules
+# options nvidia NVreg_EnableGpuFirmware=0
+
+# NVreg_EnableResizableBar=1 (Default 0) - Enable Resizable Bar support.
+# Only by enabling through here, it is also active here /proc/driver/nvidia/params
```
Comment (ventureoo) [inline usr/lib/modprobe.d/nvidia.conf:29]: This message can be remved, this applies to any parameters.

File: usr/lib/modprobe.d/nvidia.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (ventureoo) [reply usr/lib/modprobe.d/nvidia.conf:39]: I also think flickering is an XY problem. Forcing powermizer schemes seems more like a workaround than a solution.

[the remark this answers is not in the document]
Comment (dougg0k) [reply usr/lib/modprobe.d/nvidia.conf:39]: Maybe at least leave as optional there?

[the remark this answers is not in the document]
Comment (ventureoo) [reply usr/lib/modprobe.d/nvidia.conf:39]: I'm fine with it with the note that only older versions of driver may need it.

File: usr/lib/modprobe.d/nvidia.conf [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
 # open kernel modules.
-#
+
+# Disable GSP Firmware on NVIDIA's closed source kernel modules
+# options nvidia NVreg_EnableGpuFirmware=0
+
+# NVreg_EnableResizableBar=1 (Default 0) - Enable Resizable Bar support.
+options nvidia NVreg_EnableResizableBar=1
+
+# Vreg_EnableStreamMemOPs=1 (Default 0) - Activates the support for
```
Comment (ventureoo) [inline usr/lib/modprobe.d/nvidia.conf:31]:
```suggestion
# NVreg_EnableStreamMemOPs=1 (Default 0) - Activates the support for
```

**Review by ventureoo — APPROVED** [review]

**Review by Edgars-Cirulis — COMMENTED** [review]
Since powermizer is dead , maybe remove it?

**Comment by ventureoo** [comment]
> Since powermizer is dead , maybe remove it?

We may want to remove this when we stop supporting older driver versions like 390xx.

**Comment by dougg0k** [comment]
Maybe this would be another to have as optional. `NVreg_DynamicPowerManagementVideoMemoryThreshold=0`

https://www.reddit.com/r/cachyos/comments/1dw2c73/fix_found_for_nvidia_gpu_fails_to_power_off/

**Comment by dougg0k** [comment]
Added `hdmi_deepcolor` as optional option. https://www.phoronix.com/news/NVIDIA-545.23.06-Linux-Beta

**Comment by ventureoo** [comment]
> Added `hdmi_deepcolor` as optional option. https://www.phoronix.com/news/NVIDIA-545.23.06-Linux-Beta

It was enabled by default in the 555 driver.

**Comment by dougg0k** [comment]
`can be optionally enabled`

**Comment by Edgars-Cirulis** [comment]
> `can be optionally enabled`

To disable it? It's enabled by default on 555.

**Comment by Edgars-Cirulis** [comment]
> > `can be optionally enabled`
> 
> To disable it? Why. Its good feature.  It's enabled by default on 555.

**Comment by Edgars-Cirulis** [comment]
Good choice  if it doesn't support 10bit it will fallback to 8bit :)

**Comment by Edgars-Cirulis** [comment]
https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia-utils/-/commit/1b02daa2ccca6a69fa4355fb5a369c2115ec3e22

**Comment by 1Naim** [comment]
> https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia-utils/-/commit/1b02daa2ccca6a69fa4355fb5a369c2115ec3e22
> 
> We will not need anymore to enable fbdev and modeset you can make it optimally to disable it.

We've already patched this into the driver for a long time already. Furthermore, we have our own packages for NVIDIA and do not depend on Arch.

**Comment by ventureoo** [comment]
> https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia-utils/-/commit/1b02daa2ccca6a69fa4355fb5a369c2115ec3e22
> 
> We will not need anymore to enable fbdev and modeset you can make it optimally to disable it.

We still need it here for older driver versions such as 470.xx

**Comment by dougg0k** [comment]
This is not moving forward, if you decided to add, just merge, after.

**Pull request closed without merging**